ComfortZoneApp/app/src/main/java/com/example/comfortzone/Utils/LocationUtil.java

```diff
@@ -0,0 +1,96 @@
+package com.example.comfortzone.Utils;
+
+import android.Manifest;
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.location.Location;
+import android.location.LocationManager;
+import android.os.Looper;
+import android.provider.Settings;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.core.app.ActivityCompat;
+
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationCallback;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationResult;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+
+public class LocationUtil {
+
+    public static final String TAG = "locationutil";
+    public static final int PERMISSION_ID = 44;
+    public static String lat;
+    public static String lon;
+
+
+    @SuppressLint("MissingPermission")
+    public static void getLastLocation(Activity activity, Context context, FusedLocationProviderClient fusedLocationClient) {
+        if (checkPermissions(context)) {
+            if (isLocationEnabled(context)) {
+                fusedLocationClient.getLastLocation().addOnCompleteListener(new OnCompleteListener<Location>() {
+                    @Override
+                    public void onComplete(@NonNull Task<Location> task) {
+                        Location location = task.getResult();
+                        if (location == null) {
+                            requestNewLocationData(context, fusedLocationClient);
+                        } else {
+                            lat = String.valueOf(location.getLatitude());
+                            lon = String.valueOf(location.getLongitude());
+                        }
+                    }
+                });
+            } else {
+                Toast.makeText(context, "please turn on your location", Toast.LENGTH_LONG).show();
```

Comment:
We should maybe add a string resource for this message please turn on your location instead of hardcoding here.